src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java

```diff
@@ -1747,17 +1738,29 @@ public void testInputAnnotationScopeIndentationCheck() throws Exception {
         verifyWarns(checkConfig, fileName, expected);
     }
 
-    @Override
-    protected String internalGetCheckMessage(
-            String messageBundle, String messageKey, Object... arguments) {
-        final ResourceBundle resourceBundle = ResourceBundle.getBundle(
-                messageBundle,
-                Locale.ROOT,
-                Thread.currentThread().getContextClassLoader(),
-                new LocalizedMessage.Utf8Control());
-        final String pattern = resourceBundle.getString(messageKey);
-        final MessageFormat formatter = new MessageFormat(pattern, Locale.ROOT);
-        return formatter.format(arguments);
+    @Test
+    public void testPropertyOrderOfErrorMessages() {
+        final int propertyCount = 3;
+        final String[] properties = IntStream.range(0, propertyCount)
+                .mapToObj(index -> "##" + index + "##").toArray(String[]::new);
+        final String[] messages = {
+            getCheckMessage(MSG_ERROR, (Object[]) properties),
+            getCheckMessage(MSG_CHILD_ERROR, (Object[]) properties),
+            getCheckMessage(MSG_ERROR_MULTI, (Object[]) properties),
+            getCheckMessage(MSG_CHILD_ERROR_MULTI, (Object[]) properties),
+        };
+        final boolean isInOrder = Arrays.stream(messages).allMatch(msg -> {
+            final int[] indices = Arrays.stream(properties).mapToInt(msg::indexOf).toArray();
+            int prev = -1;
+            for (int index : indices) {
```

Comment:
I am probably biased, as I don't use Java 8 lambdas and think of some of their uses as not easy to read.
Your original implementation is definitely too complex, imo. I would still prefer the loop on properties.